src/Cassandra/Outputs/OutputRows.cs

```diff
@@ -17,33 +17,30 @@
 using System;
 using System.Collections.Generic;
 using System.Threading;
-// ReSharper disable CheckNamespace
 
+// ReSharper disable CheckNamespace
 namespace Cassandra
 {
     internal class OutputRows : IOutput
     {
-        public readonly int RowLength;
-        private readonly Guid? _traceId;
-        private RowSetMetadata _metadata;
-        private byte _protocolVersion;
+        private readonly int _rowLength;
+        private readonly RowSetMetadata _metadata;
+        private readonly byte _protocolVersion;
+        private static readonly ThreadLocal<byte[]> ReusableBuffer = new ThreadLocal<byte[]>(() => new byte[16]);
```

Comment:
We need a buffer per thread, as the same thread can not be reading an output at the same time. There can be thousands of `OutputRows` instances at any given time, having a static thread local saves a lot of allocations.